### DISCUSSION TITLE
Experiment_Tracking_Train_Loop_Stuck

### DISCUSSION THREAD

**Discussion by sicajc** [mrdbourke/pytorch-deep-learning#116] [Q&A]
I am having trouble running the code of Pytorch Model07 experiment tracking. When I am running the train helper function, it seems like it just get stuck, nothing is running and I have no idea what is going on. Please help! Thanks. I'm running the code on the interactive window of Jupyter Notebook in VScode.

![train_problem](https://user-images.githubusercontent.com/58667661/187078968-c96632bf-ef03-4181-b089-46eaf8e4dabf.png)

**Comment by mrdbourke** [comment]
Hi @sicajc,

Let's figure this out.

What have you tried so far?

Is the code you're running tracking experiments with TensorBoard? Potentially you'll have to install the TensorBoard plugin for VS Code - https://code.visualstudio.com/docs/datascience/pytorch-support#_tensorboard-integration 

Have you tried to restart the notebook and run the cell(s) again?

**Reply by sicajc** [reply]
I have Tensorboard downloaded and restarted the cells, but it seems like it is not working.


![problem4](https://user-images.githubusercontent.com/58667661/187134662-8c100668-8319-4bb8-bd58-63bc45191518.png)

**Comment by sicajc** [comment]
Oh~ I suppose I figured out the bug. It seems like on my cpu, when creating dataloader with num_workers != 0 , the dataLoader is not working as expected. I tested it out by using next(iter(dataloader)), and it just went into an infinite loop and never stop running. Then i have no idea how to enable the use of multiple cpus when loading dataLoader if that is the case? Thx

**Reply by sicajc** [reply]
If I set num_of_workers = 0 , thing works fine
